validator/src/main.rs

```diff
@@ -1394,9 +1395,12 @@ pub fn main() {
 
     let maximum_local_snapshot_age = value_t_or_exit!(matches, "maximum_local_snapshot_age", u64);
     let maximum_full_snapshot_archives_to_retain =
-        value_t_or_exit!(matches, "maximum_full_snapshots_to_retain", usize);
-    let maximum_incremental_snapshot_archives_to_retain =
-        value_t_or_exit!(matches, "maximum_incremental_snapshots_to_retain", usize);
+        value_t_or_exit!(matches, "maximum_full_snapshots_to_retain", NonZeroUsize);
```

Comment:
As-is, the error emitted here will be:
```
error: Invalid value: The argument '0' isn't a valid value
```
I could see this being kind of confusing, and I could do `value_t!()`, map the result and spit out a more specific error message (ie naming the argument); thoughts?

Reply:
Yeah, that works for me. I was initially looking/thinking of using `.validator()` when we discussed this issue last week; not sure why I went away from it but I like that it gets the boilerplate logic out of the main function